expression/typeinferer_test.go

```diff
@@ -177,6 +177,8 @@ func (ts *testTypeInferrerSuite) TestInferType(c *C) {
 		{"weekday('2009-12-31 23:59:59.000010')", mysql.TypeLonglong, charset.CharsetBin, mysql.BinaryFlag},
 		{"weekofyear('2009-12-31 23:59:59.000010')", mysql.TypeLonglong, charset.CharsetBin, mysql.BinaryFlag},
 		{"yearweek('2009-12-31 23:59:59.000010')", mysql.TypeLonglong, charset.CharsetBin, mysql.BinaryFlag},
+		{"addtime('2007-12-31 23:59:59.999999', '1 1:1:1.000002')", mysql.TypeDatetime, charset.CharsetBin, mysql.BinaryFlag},
```

Comment:
'2009-12-31 23:59:59.000010' is a string but not a datetime,
you can use c_datetime (which can be found in line51).